packages/core/src/actions/execute.ts

```diff
@@ -1137,6 +1137,62 @@ export const sortSigners = (arr: Array<TreeSigner>): void => {
   })
 }
 
+/**
+ * This is a utility function that is helpful for recovering from a deployment that has timed out. It fetches
+ * all the transaction receipts related to a deployment using the expected events and returns them. If this
+ * function is used when some of the transaction receipts are already tracked, they can be passed in and this
+ * function will return an array of all of the unique transactions.
+ *
+ * This function returns standard ethers.TransactionReceipt objects instead of our SphinxTransactionReceipt
+ * type because that is more flexible and generally useful.
+ */
```

Comment:
nit: May be worth mentioning that this function modifies the given `receipts` array